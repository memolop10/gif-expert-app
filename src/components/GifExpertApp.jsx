import React,{useState} from 'react'
import AddCategory from './AddCategory';
import GifGrid from './GifGrid';

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch']);

    return (
        <>
            <h2> GifExpertApp </h2>
            <AddCategory setCategories={ setCategories }/>
            <hr />
        {
            categories.map( category => (
                <GifGrid
                    key={ category }
                    category={ category }
                />
            ))  
        }   
        </>
    )
}

export default GifExpertApp
