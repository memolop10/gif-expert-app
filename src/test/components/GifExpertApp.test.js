import React from 'react'
import { shallow } from "enzyme";
import GifExpertApp from "../../components/GifExpertApp";


describe('Test in component GifExpertApp', () => {
    test('should render the component correctly', () => {
        let wrapper = shallow( <GifExpertApp/> )
        expect(wrapper).toMatchSnapshot()
    })
    
});
