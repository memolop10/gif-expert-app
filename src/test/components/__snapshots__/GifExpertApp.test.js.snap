// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in component GifExpertApp should render the component correctly 1`] = `
<Fragment>
  <h2>
     GifExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
